### ISSUE TITLE
 Issue loading onnx model to wasm using `onnxruntime-web`

### ISSUE THREAD

**Issue by ailen-wrx** [microsoft/onnxruntime#11413]
**Describe the bug**
I have read #9322 and followed the steps. Following the steps I tried to load an onnx model to wasm instead of webgl (because my model have some layers which cannot be resolved on webgl). I got the error `Error: no available backend found. ERR: [wasm] RuntimeError: Aborted(CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0). Build with -s ASSERTIONS=1 for more info.`

**Urgency**
I wish I can solve this problem by **mid May** for presentation.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- ONNX Runtime installed from (source or binary): binary
- ONNX Runtime version: latest
- Python version: 3.6.9
- CUDA/cuDNN version: 11.4

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.

I have a project built by vue.js. I have installed onnxruntime-web and saved into package.json:
```JavaScript
    "onnxruntime": "^1.7.0",
    "onnxruntime-node": "^1.11.0",
    "onnxruntime-web": "^1.11.0",
```

I added the plugin to `vue-config.js`:
```JavaScript
const CopyWebpackPlugin = require('copy-webpack-plugin')
module.exports = {
  configureWebpack: {
    module: {
      rules: [
        {
          test: /\.txt$/i,
          use: 'raw-loader',
        }
      ]
    },
    plugins: [
      new CopyWebpackPlugin([
        {
          from: "node_modules/onnxruntime-web/dist/*.wasm",
          to  : "[name][ext]"
        },
    ])
    ]
  },
  devServer: {
    proxy: {
      '^/filter-api': {
  ...
```

I tried to load the model in a JavaScript source file:
```JavaScript
import { InferenceSession } from 'onnxruntime-web'
...

const modelUrl = '/onnx/ai_sharp.onnx'
await InferenceSession.create(modelUrl, { executionProviders: ['wasm'], })
...
```

The model is actually the generation model in [CycleGAN](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix):
```python
        model = [nn.ReflectionPad2d(3),
                 nn.Conv2d(input_nc, ngf, kernel_size=7, padding=0, bias=use_bias),
                 norm_layer(ngf),
                 nn.ReLU(True)]

        n_downsampling = 2
        for i in range(n_downsampling):  # add downsampling layers
            mult = 2 ** i
            model += [nn.Conv2d(ngf * mult, ngf * mult * 2, kernel_size=3, stride=2, padding=1, bias=use_bias),
                      norm_layer(ngf * mult * 2),
                      nn.ReLU(True)]

        mult = 2 ** n_downsampling
        for i in range(n_blocks):       # add ResNet blocks

            model += [ResnetBlock(ngf * mult, padding_type=padding_type, norm_layer=norm_layer, use_dropout=use_dropout, use_bias=use_bias)]

        for i in range(n_downsampling):  # add upsampling layers
            mult = 2 ** (n_downsampling - i)
            model += [nn.ConvTranspose2d(ngf * mult, int(ngf * mult / 2),
                                         kernel_size=3, stride=2,
                                         padding=0, output_padding=0,
                                         bias=use_bias),
                      norm_layer(int(ngf * mult / 2)),
                      nn.ReLU(True)]
        model += [nn.ReflectionPad2d(3)]
        model += [nn.Conv2d(ngf, output_nc, kernel_size=7, padding=0)]
        model += [nn.Tanh()]

        self.model = nn.Sequential(*model)
```

When I ran the server and opened in Chrome, I found in the developer tool that these files are fetched:
<img width="438" alt="Screen Shot 2022-05-01 at 23 59 06" src="https://user-images.githubusercontent.com/67893756/166154008-c2745e4b-f429-4dbc-9463-93490420e126.png">
<img width="441" alt="Screen Shot 2022-05-02 at 00 00 35" src="https://user-images.githubusercontent.com/67893756/166154077-42f145e5-23fa-4fc5-a29d-d47d018e740b.png">

But in the console I got the errors:
```
wasm streaming compile failed: TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.

falling back to ArrayBuffer instantiation

failed to asynchronously prepare wasm: CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0

ort-web.min.js?f1d4:6 Aborted(CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0)
```

Back to the JavaScript source file, when I loaded the model with normal `webgl` backend, I got `TypeError: cannot resolve operator 'ConvTranspose' with opsets: ai.onnx v9`, which is expected becaue this operation is not supported.

**Expected behavior**
Successfully load the model on wasm backend.

**Comment by fs-eire** [comment]
"3c 21 44 4f" is the ASCII code of "<!DO", which is usually a HTML page instead of content of a .wasm file.

From the response header, we can see the content type is text/html. This is not expected. Possible reasons are:
- the file path for the .wasm file is incorrect. make sure put it in the same directory to your bundle js file to serve.
- the server does not support web assembly MIME type (application/wasm). Use a server that supports serving wasm file. (for example NPM package `http-server` does not support it. Use `light-server` if this is your case)

**Comment by ailen-wrx** [comment]
> "3c 21 44 4f" is the ASCII code of "<!DO", which is usually a HTML page instead of content of a .wasm file.
> 
> From the response header, we can see the content type is text/html. This is not expected. Possible reasons are:
> 
> * the file path for the .wasm file is incorrect. make sure put it in the same directory to your bundle js file to serve.
> * the server does not support web assembly MIME type (application/wasm). Use a server that supports serving wasm file. (for example NPM package `http-server` does not support it. Use `light-server` if this is your case)

Thanks! Let me look into the code and update here when I solve it.

**Comment by ailen-wrx** [comment]
`vuetify-loader` cannot resolve the latest version of `webpack` bundle higher than `4.0.0`, while `CopyWebpackPlugin` requires a `webpack` version higher than that. Issue solved by force installing `webpack@4.0.0` and manually copying the *.wasm files into `./public/js`.
